### PR TITLE
Add test examples for non-ideal situations

### PR DESCRIPTION
**Pull request by snowe2010**

### CODE REVIEW

File: core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt [hunk 1]
```diff
@@ -587,6 +587,116 @@ class GoogleStyleFormatterKtTest {
               .trimMargin(),
       )
 
+    @Test
+    fun `Trailing comma forces variable value in list onto new line with manageTrailingCommas turned off`() =
```
Comment (davidtorosyan) [inline core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt:591]: Move to `FormatterTest.kt`, omit `FormattingOptions`

File: core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt [hunk 2]
```diff
@@ -587,6 +587,116 @@ class GoogleStyleFormatterKtTest {
...
+            formattingOptions = FormattingOptions(
+                blockIndent = 4,
+                continuationIndent = 4,
+                manageTrailingCommas = false
+            ),
+            deduceMaxWidth = false,
+        )
+
+    @Test
+    fun `nested functions, maps, and statements in named parameters`() =
```
Comment (davidtorosyan) [inline core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt:611]: Move to `FormatterTest.kt`, omit `formattingOptions ` if possible

File: core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt [hunk 3]
```diff
@@ -587,6 +587,116 @@ class GoogleStyleFormatterKtTest {
...
+                    }
+            )
+
+            """.trimIndent(),
+            formattingOptions = Formatter.KOTLINLANG_FORMAT,
+            deduceMaxWidth = true,
+        )
+    
+    @Test
+    fun `complex calls and calculation in named parameters with wrapping`() =
```
Comment (davidtorosyan) [inline core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt:634]: Move to `FormatterTest.kt`, omit `formattingOptions ` if possible

File: core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt [hunk 4]
```diff
@@ -587,6 +587,116 @@ class GoogleStyleFormatterKtTest {
...
+                denominator = r * (BigDecimal.ONE + r).pow(n),
+            )
+
+            """.trimIndent(),
+            formattingOptions = Formatter.KOTLINLANG_FORMAT,
+            deduceMaxWidth = true,
+        )
+
+    @Test
+    fun `complex calls and calculation in named parameters without wrapping`() =
```
Comment (davidtorosyan) [inline core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt:656]: Move to `FormatterTest.kt`, omit `formattingOptions ` if possible

File: core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt [hunk 5]
```diff
@@ -587,6 +587,116 @@ class GoogleStyleFormatterKtTest {
...
+                denominator = r * (BigDecimal.ONE + r).pow(n),
+            )
+
+            """.trimIndent(),
+            formattingOptions = Formatter.KOTLINLANG_FORMAT,
+            deduceMaxWidth = false,
+        )
+
+    @Test
+    fun `long call chains in named parameters`() =
```
Comment (davidtorosyan) [inline core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt:672]: Remove `formattingOptions`, this file automatically picks up google style

**Review by davidtorosyan — CHANGES_REQUESTED** [review]
Thanks for adding more tests! I left some comments about moving them to the right places / adjusting the options. This is mainly to help with maintainability (the more tests have the same set of options, the easier to reason about).

File: core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt (hunk 1 above)
Comment (snowe2010) [reply core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt:591]: Hm. but the test is showing the bug with the formatting options. If I omit that, the test will act differently. I've gone ahead and moved it to the FormatterTest, but left the FormattingOptions in, else the test is pointless.

File: core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt (hunk 5 above)
Comment (snowe2010) [reply core/src/test/java/com/facebook/ktfmt/format/GoogleStyleFormatterKtTest.kt:591]: done.

**Review by snowe2010 — COMMENTED** [review]
@davidtorosyan I also added one more test around `if` expression functions

**Comment by hick209** [comment]
We apologize for the delay. Let's try to get another look into this

**Pull request closed without merging**